documentation/src/docs/state-synchronizer-design.md

```diff
@@ -7,24 +7,28 @@ You may obtain a copy of the License at
 
-Imagine you want many processes to share a Map. This can be done by creating by a StateSynchronizer, which will help coordinate changes to the map. Each client has their own copy of the map in memory and can apply updates by passing a generator to the StateSynchronizer. Every time an update is attempted the updated is recorded to a segment. Updates will fail unless the Map passed into the update method is consistent with all of the updates that have been recorded to the segment. If this occurs the generator is called with the latest state to try again. Thus the order of updates is defined by the order in which they are written to the segment.
+A concrete [example](https://github.com/pravega/pravega/blob/master/client/src/test/java/io/pravega/client/state/examples/SetSynchronizer.java) of synchronizing the contents of a set is provided. We also have an example that is synchronizing [membership of a set of hosts](https://github.com/pravega/pravega/blob/master/client/src/test/java/io/pravega/client/state/examples/MembershipSynchronizer.java).
 
-# How this is implemented
-For this to work we use two features of the Pravega Segment Store Service.
-## Conditional append
-The append method can specify what offset the append is expected to be at. If the append will not go in, nothing should be done and a failure should be returned to the client.
+Imagine you want many processes to share a Map. This can be done by creating the State Synchronizer, it will aid in coordinating the changes to the Map. Each client has their own copy of the Map in memory and can apply updates by passing a generator to the State Synchronizer. Every time an update is attempted the updates are recorded to the Stream Segment. Updates are successful when the Map passed into the update method is consistent with all of the updates that have been recorded to the Stream Segment. If this occurs the generator is called with the latest state to try again. Thus the order of updates is defined by the order in which they are written to the Stream Segment.
 
-## Truncate segment
-Truncate segment deletes all data before a given offset. (This operation does not affect the existing offsets)
-Any reads for offsets lower than this value will fail. Any data stored below this offset can be removed.
+# Implementation
+
+For the implementation, two features of the Pravega Segment Store Service is used.
+
+## Conditional Append
+
+The append method can specify what offset the append is expected to be at. If the append will not go in, a fail message should be returned to the client.
+
+## Truncate Segment
+Truncate Segment deletes all data before a given offset. (This operation does not affect the existing offsets)
```

Comment:
I think that the sentence in parenthesis could be without parenthesis.

Reply:
Similarly to the previous comment, I think that we need to explain here that Segment Truncation is used for compaction in the State Synchronizer, as the log formed by old update operations that does not need to be kept forever. If we do not introduce that, the paragraph lacks from context.

Reply:
This is not the only use of truncation. It is also used when we set a retention policy to remove old data. I don't think that needs to be mentioned here.

Reply:
@tkaitchuck  Accepted suggestion